From d080fd514675c5e8ad6add7e93b23ed0641365d7 Mon Sep 17 00:00:00 2001
From: Matthias Brugger <mbrugger@suse.com>
Date: Mon, 27 Jun 2016 18:49:02 +0200
Subject: arm64: acpi/numa: Add bad_srat() and srat_disabled()
 stub
Git-commit: e84025e274e66986277e11f0dda03373e246ffad (partial)
Patch-mainline: v4.8-rc1
References: fate#319981

Add bad_srat() and srat_diabled() stubs.

Based on:
e84025e274e6 ("ACPI / NUMA: move bad_srat() and srat_disabled() to drivers/acpi/numa.c")

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
The patch is reworked to not touch generic code

 arch/arm64/kernel/acpi_numa.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/kernel/acpi_numa.c b/arch/arm64/kernel/acpi_numa.c
index 45aaa7f..ff3573c 100644
--- a/arch/arm64/kernel/acpi_numa.c
+++ b/arch/arm64/kernel/acpi_numa.c
@@ -29,6 +29,7 @@
 #include <asm/numa.h>
 
 static int cpus_in_srat;
+int acpi_numa __initdata;
 
 struct __node_cpu_hwid {
 	u32 node_id;    /* logical node containing this CPU */
@@ -38,6 +39,17 @@ struct __node_cpu_hwid {
 static struct __node_cpu_hwid early_node_cpu_hwid[NR_CPUS] = {
 [0 ... NR_CPUS - 1] = {NUMA_NO_NODE, PHYS_CPUID_INVALID} };
 
+void __init bad_srat(void)
+{
+	pr_err("SRAT: SRAT not used.\n");
+	acpi_numa = -1;
+}
+
+int __init srat_disabled(void)
+{
+	return acpi_numa < 0;
+}
+
 /*
  * Callback for SLIT parsing.  pxm_to_node() returns NUMA_NO_NODE for
  * I/O localities since SRAT does not list them.  I/O localities are
-- 
2.8.4

